Ensure all strings in block pattern templates are translated and escaped

alexsanford opened the pull request:
Fixes #6321 

### Changes proposed in this Pull Request

Set up translations in the Course List block pattern template.

### Testing instructions

1. Set your site's language to Spanish (note that you may need to save Permalinks to flush the cache and get the frontend paths working).
2. Download the latest translations on the Dashboard > Updates page (Escritorio > Actualizaciones)
3. Create three courses. Add two lessons to each.
4. On one course, complete one lesson.
5. On another course, complete all lessons.
6. Do not start the last course.
7. Install Loco Translate. Open the translation file for Sensei LMS in the language you've chosen.
8. Add translations for "Start Course", "Continue", and "Visit Results" and save. This will ensure the `.po`/`.mo` and `.json` files are generated.
9. Create a new page and add the Course List block (Lista del curso).
10. Ensure that in the editor, the buttons are translated (Empezar el curso)
11. Publish the page and view the frontend. You should see all three button statuses, and they should all be translated: Empezar el curso, Continuar, Visitar resultados.

alexsanford committed:
Ensure all strings in block pattern templates are translated and escaped

alexsanford commented on includes/block-patterns/course-list/class-sensei-course-list-block-patterns.php:
```diff
@@ -166,11 +166,11 @@ public function register_course_list_block_patterns() {
 						<div class="wp-block-group alignfull sensei-course-theme-course-list-pattern has-background-color has-primary-background-color has-text-color has-background has-link-color" style="padding-top:5rem;padding-right:20px;padding-bottom:100px;padding-left:20px"><!-- wp:group {"style":{"border":{"left":{"color":"var:preset|color|background","width":"1px"}},"spacing":{"padding":{"left":"20px"},"mar ...
 						<div class="wp-block-group" style="border-left-color:var(--wp--preset--color--background);border-left-width:1px;margin-bottom:40px;padding-left:20px"><!-- wp:heading {"level":2,"style":{"typography":{"textTransform":"uppercase","lineHeight":"1"},"spacing":{"padding":{"top":"0px"}}},"textColor":"background","className":"sensei-pattern-heading","fontSize":"medium"} -->
-						<h2 class="sensei-pattern-heading has-background-color has-text-color has-medium-font-size" style="padding-top:0px;line-height:1;text-transform:uppercase"><strong>' . __( 'Course list', 'sensei-lms' ) . '</strong></h2>
+						<h2 class="sensei-pattern-heading has-background-color has-text-color has-medium-font-size" style="padding-top:0px;line-height:1;text-transform:uppercase"><strong>' . esc_html__( 'Course list', 'sensei-lms' ) . '</strong></h2>
```
I added HTML escaping to some translated strings which are not directly related to this PR. This should be a safe change to add a bit of extra security, but if it seems dangerous for some reason I can remove these changes.

renatho reviewed (commented):
I tested it in Jurassic Ninja using Español and Español de México, and in both, I just saw the translation in the "Continue".

I suspect it's because the translation is not coming for these texts in these languages. But at the same time, in a quick check in translate.wordpress.org, I see the translations.

Tomorrow I'll check with more details, but I'd like to confirm if you used a different Spanish language.

alexsanford commented:
Hmm. I'll test in JN. It's working locally, and I also noticed that .org has the translations for those strings. Maybe something still isn't quite right. Thanks for testing!

alexsanford commented:
I've confirmed the issue, and per our Slack conversation I'll work on a fix. Will update here when finished!

alexsanford committed:
Regenerate POT file for testing

alexsanford commented:
I've updated the POT file (https://github.com/Automattic/sensei/pull/6322/commits/44eaeb924661b39cf419e5068c889f9512896fc0) and also updated the instructions to include the following steps:

- Install Loco Translate. Open the translation file for Sensei LMS in the language you've chosen.
- Add translations for "Start Course", "Continue", and "Visit Results" and save. This will ensure the .po/.mo and .json files are generated.

Tested again on JN and it worked! 🎉

renatho reviewed (approved):
Thank you for checking! Tested with Loco and it's working properly! 🎉
